Cython/Utility/ModuleSetupCode.c

```diff
@@ -1588,15 +1588,28 @@ static int __Pyx_check_binary_version(void);
 /////////////// CheckBinaryVersion ///////////////
 
 static int __Pyx_check_binary_version(void) {
-    char ctversion[4], rtversion[4];
-    PyOS_snprintf(ctversion, 4, "%d.%d", PY_MAJOR_VERSION, PY_MINOR_VERSION);
-    PyOS_snprintf(rtversion, 4, "%s", Py_GetVersion());
-    if (ctversion[0] != rtversion[0] || ctversion[2] != rtversion[2]) {
+    char ctversion[5], rtversion[5] = {'\0'};
```

Comment:
Is `rtversion[5]` correct with `0 <= i < 4`? Since you seem to be using it straight in snprintf() below, I think you need to at least append a `\0`, no?

Reply:
`rtversion[5] = {'\0'};` zero initializes the whole thing I believe (you only need to specify the first zero) so it should be fine